src/cdi/CPEloss.hh

```diff
@@ -129,7 +132,40 @@ public:
 	 * \brief Returns the general Eloss model type (Analytic),
 	 * defined in the enum at the top of this file.
 	 */
-  virtual rtt_cdi::CPModelType getModelType() const = 0;
+  //virtual rtt_cdi::CPModelType getModelType() const = 0;
+
+  /*!
+   * \brief Query to determine the transporting particle type.
+   */
+  CParticle getProjectile() const { return projectile; }
```

Comment:
What do you think about making projectile a public data member?  This would allow us to eliminate these _getters_ from the class implementation.  Also consider if you want to return a reference to the _projectile_ or a copy.

Reply:
My thinking was to ensure the target/projectile members cannot be modified after construction, is a getter function an OK way to indicate this? Also, from the discussion about passing target/projectile by copy rather than reference, should it also be a copy that gets returned?